core/templates/services/prevent-page-unload-event.service.ts

```diff
@@ -25,35 +25,46 @@ import { WindowRef } from 'services/contextual/window-ref.service';
 })
 export class PreventPageUnloadEventService {
   private listenerActive: boolean;
+  validation: Function;
+  _preventPageUnloadEventHandlerBind = undefined;
   constructor(private windowRef: WindowRef) {
     this.listenerActive = false;
+    this.validation = undefined;
   }
 
-  addListener(): void {
+  addListener(callback: Function = () => {
+    return true;
+  }): void {
```

Comment:
Why is the type definition a function that returns true? Is this accurate?

Reply:
If a callback is passed as an argument then, the alert will be displayed only when the page is reloaded and the callback returns true.
If a callback is not passed as an argument then, The alert will be displayed whenever the page is reloaded.

Reply:
> Why is the type definition a function that returns true? Is this accurate?

```
callback: Function = () => {
    return true;
  }
```
Here the type definition is a `Function` and the default values is `() => {return true;}`

Reply:
> If a callback is passed as an argument then, the alert will be displayed only when the page is reloaded and the callback returns true.
> If a callback is not passed as an argument then, The alert will be displayed whenever the page is reloaded.

Sounds like callback can either be a function or not provided at all and that when it is a function it will return a boolean value? If so would the "type definition" look something like this:

`callback?: () => boolean`

Reply:
Please comment "done" or similar if this has been addressed.

Reply:
DONE